components/script/dom/htmlvideoelement.rs

```diff
@@ -280,11 +316,32 @@ impl VirtualMethods for HTMLVideoElement {
     fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
         self.super_type().unwrap().attribute_mutated(attr, mutation);
 
-        if let Some(new_value) = mutation.new_value(attr) {
-            if attr.local_name() == &local_name!("poster") {
-                self.fetch_poster_frame(&new_value, CanGc::note());
-            }
-        }
+        match *attr.local_name() {
+            local_name!("poster") => match mutation {
+                AttributeMutation::Set(_) => {
+                    if let Some(new_value) = mutation.new_value(attr) {
+                        self.fetch_poster_frame(&new_value, CanGc::note())
+                    }
+                },
+                AttributeMutation::Removed => {
+                    self.set_intrinsic_size();
+                    self.htmlmediaelement.clear_current_frame();
+                },
+            },
```

Comment:
`mutation.new_value(attr)` being `Some(_)` seems equivalent to checking `AttributeMutation::Set(_)`.

```suggestion
            local_name!("poster") => {
                if let Some(new_value) = mutation.new_value(attr) {
                    self.fetch_poster_frame(&new_value, CanGc::note())
                } else {
                    self.set_intrinsic_size();
                    self.htmlmediaelement.clear_current_frame();
                }
            },
```